source/pages/Settings/ExternalWatchAsset.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState } from 'react';
+
+import {
+  DefaultModal,
+  Layout,
+  LoadingComponent,
+  PrimaryButton,
+  SecondaryButton,
+} from 'components/index';
+import { useQueryData, useUtils } from 'hooks/index';
+import { ITokenEthProps } from 'types/tokens';
+import { dispatchBackgroundEvent, getController } from 'utils/browser';
+
+const ExternalWatchAsset = () => {
+  const { host, ...data } = useQueryData();
+  const [loading, setLoading] = useState(false);
+  const [confirmed, setConfirmed] = useState(false);
+  const [assetInfo, setAssetInfo] = useState<ITokenEthProps>();
+  const { alert } = useUtils();
+  const controller = getController();
+  const wallet = controller.wallet;
+
+  function formatAssetData(asset: any) {
```

Comment:
This specific function is used here because the method `wallet_watchAsset` can send an object or an array, it's a particular scenario. But the others methods sends an array as standard.